sale_order_price_recalculation/models/sale_order.py

```diff
@@ -0,0 +1,42 @@
+# Copyright 2014 Carlos Sánchez Cifuentes <csanchez@grupovermon.com>
+# Copyright 2015 Pedro M. Baeza <pedro.baeza@serviciosbaeza.com>
+# Copyright 2015 Oihane Crucelaegui <oihanecrucelaegi@avanzosc.es>
+# Copyright 2016 Vicent Cubells <vicent.cubells@tecnativa.com>
+# Copyright 2017 David Vidal <david.vidal@tecnativa.com>
+# Copyright 2020 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+from odoo import models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    def _recalculate_prices_get_fields(self):
+        return ["price_unit", "discount"]
```

Comment:
@rousseldenis Your change to use 'onchange_helper' LGTM nevertheless can you add a comment to explain the purpose of this methods and where these fields are used.

Reply:
Please don't use `onchange_helper` and add a new dependency. You just need to use core `odoo.tests.Form`, that makes the same. Example:

https://github.com/OCA/sale-workflow/pull/1157/commits/458ef3a4796b935d3e05f79b7a0481a592a43d6b

Reply:
I am reluctant to use a method written for testing in business code. Especially since it degrades performance quite strongly.

Reply:
It's already used in business code in Odoo itself, and it degrades the same as `onchange_helper`, as it triggers onchanges.

Reply:
I was sure you want to use it but I agree with @lmignon. That's not because Odoo uses it in their code, that's a good idea...

Reply:
Moreover, the import speaks by itself... 

`from odoo.tests.common import Form`

Reply:
The only thing I found is that product.price is not in tocompute, so this is the cache value which is used (wrong one).

The only 'hack' I found is to add price field. But this is not sustainable as we cannot figure out which new computed field could be added to the onchange.

But as we know which one we want to modify, onchange_helper is definetely the solution.

![image](https://user-images.githubusercontent.com/19529533/83883342-a3bde500-a743-11ea-8945-2250e0376665.png)

Reply:
About the `Form` being in `odoo.tests`, it's because historical reasons, starting as a helper for tests, but then seeing a convenient way to populate all values, and spreading its use across business code. You can see several examples of usage in latest developed modules:

https://github.com/odoo/odoo/blob/2b4b2ce8660326b1cb53b5624690718b89aaceda/addons/account_facturx/models/account_move.py#L99
https://github.com/odoo/odoo/blob/2b4b2ce8660326b1cb53b5624690718b89aaceda/addons/l10n_be_edi/models/account_invoice.py#L49
https://github.com/odoo/odoo/blob/2b4b2ce8660326b1cb53b5624690718b89aaceda/addons/l10n_it_edi/models/account_invoice.py#L361

The problem you mention you have in `onchange_helper` may not happen with this helper.

The reason I prefer it is for reducing maintenance burden of the extra module and having to deal with different logics depending on the source.

Luckily, both things will go away in v14 with the refactoring to computed writeable fields, but meanwhile...

Would you accept an alternate PR using the method I comment to compare?

Reply:
> About the `Form` being in `odoo.tests`, it's because historical reasons, starting as a helper for tests, but then seeing a convenient way to populate all values, and spreading its use across business code. You can see several examples of usage in latest developed modules:
> 
> https://github.com/odoo/odoo/blob/2b4b2ce8660326b1cb53b5624690718b89aaceda/addons/account_facturx/models/account_move.py#L99
> https://github.com/odoo/odoo/blob/2b4b2ce8660326b1cb53b5624690718b89aaceda/addons/l10n_be_edi/models/account_invoice.py#L49
> https://github.com/odoo/odoo/blob/2b4b2ce8660326b1cb53b5624690718b89aaceda/addons/l10n_it_edi/models/account_invoice.py#L361
> 
> The problem you mention you have in `onchange_helper` may not happen with this helper.
> 
> The reason I prefer it is for reducing maintenance burden of the extra module and having to deal with different logics depending on the source.
> 
> Luckily, both things will go away in v14 with the refactoring to computed writeable fields, but meanwhile...
> 
> Would you accept an alternate PR using the method I comment to compare?

IMHO, onchange_helper would not be such a burden as already migrated and tested so much in previous versions. As you said, for v14, it will disappear, so... The reason whyForm() as been used in code is not clear.

As said in https://github.com/odoo/odoo/pull/25508#pullrequestreview-425975320, you cannot rely logic on that Form() implementation. I'm not a fool.

Reply:
Why can't you rely on it? It's totally trustable. And I can say that because it's used in tests as well as in logic. I have been using it for a while now without too much problems. I'm not saying that in v14 it will disappear. What will disappear is the need to use it (the same as `onchange_helper`), due to the move to computed writeable, but meanwhile in v12/v13, I prefer to use SSF (acronym of Server Side Form, which is the official name for this) than to add a new piece in the puzzle through the OCA module, that has had also problems in the past and I remember a PR that lasted a lot.